File: VB/TestPersistentRepository/RepositoryItemMyPopupContainerEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Drawing
Imports System.Collections
Imports System.Drawing





Namespace DevExpress.MyControl
	Public Class RepositoryItemMyPopupContainerEdit
		Inherits RepositoryItemPopupContainerEdit
		Private cacheSize_Renamed As Hashtable

		Friend Const EDITORTypeName_Renamed As String = "MyPopupContainerEdit"
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EDITORTypeName_Renamed
			End Get
		End Property



		<Browsable(False)> _
		Public Shadows ReadOnly Property OwnerEdit() As MyPopupContainerEdit
			Get
				Return TryCast(MyBase.OwnerEdit, MyPopupContainerEdit)
			End Get
		End Property


		Public ReadOnly Property CacheSize() As Hashtable
			Get
				Return cacheSize_Renamed
			End Get
		End Property




		Shared Sub New()
			RegisterMyPopupContainerEdit()
		End Sub



		Public Sub New()
			MyBase.New()
			cacheSize_Renamed = Nothing
			cacheSize_Renamed = New Hashtable()
		End Sub


        Public Shared Sub RegisterMyPopupContainerEdit()
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EDITORTypeName_Renamed, GetType(MyPopupContainerEdit), GetType(RepositoryItemMyPopupContainerEdit), GetType(PopupContainerEditViewInfo), New ButtonEditPainter(), True, CType(Nothing, Image), GetType(DevExpress.Accessibility.ButtonEditAccessible)))
        End Sub



		Public Overrides Sub Assign(ByVal item As RepositoryItem)
			Dim source As RepositoryItemMyPopupContainerEdit = TryCast(item, RepositoryItemMyPopupContainerEdit)
			BeginUpdate()
			Try
				MyBase.Assign(item)
				Me.cacheSize_Renamed = source.CacheSize
			Finally
				EndUpdate()
			End Try
		End Sub
	End Class
End Namespace

File: CS/TestPersistentRepository/RepositoryItemMyPopupContainerEdit.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Drawing;
using System.Collections;




namespace DevExpress.MyControl
{
    public class RepositoryItemMyPopupContainerEdit : RepositoryItemPopupContainerEdit
    {
        private Hashtable cacheSize;

        internal const string EDITORTypeName = "MyPopupContainerEdit";
        public override string EditorTypeName
        {
            get
            {
                return EDITORTypeName;
            }
        }



        [Browsable(false)]
        public new MyPopupContainerEdit OwnerEdit
        {
            get
            {
                return base.OwnerEdit as MyPopupContainerEdit;
            }
        }


        public Hashtable CacheSize
        {
            get 
            {
                return cacheSize;
            }
        }




        static RepositoryItemMyPopupContainerEdit()
        {
            RegisterMyPopupContainerEdit();
        }



        public RepositoryItemMyPopupContainerEdit()
            : base()
        {
            cacheSize = null;
            cacheSize = new Hashtable();            
        }


        public static void RegisterMyPopupContainerEdit()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EDITORTypeName,
                                                       typeof(MyPopupContainerEdit), typeof(RepositoryItemMyPopupContainerEdit),
                                                       typeof(PopupContainerEditViewInfo), new ButtonEditPainter(), true, null));
        }



        public override void Assign(RepositoryItem item)
        {
            RepositoryItemMyPopupContainerEdit source = item as RepositoryItemMyPopupContainerEdit;
            BeginUpdate();
            try
            {
                base.Assign(item);
                this.cacheSize = source.CacheSize;
            }
            finally
            {
                EndUpdate();
            }
        }
    }
}
